File: s3-source-connector/src/main/java/sourceconnector/domain/batch/DefaultMessageBatchLogs.java
package sourceconnector.domain.batch;

import lombok.RequiredArgsConstructor;

import java.util.List;

@RequiredArgsConstructor
public class DefaultMessageBatchLogs implements MessageBatch {
  private final List<String> logs;

  public void add(String log) {
    this.logs.add(log);
  }
  public void addAll(List<String> logs) {
    this.logs.addAll(logs);
  }

  @Override
  public List<String> get() {
    if (logs.isEmpty()) {
      throw new IllegalStateException("No log found");
    }
    return this.logs;
  }
}

File: source-connector/src/main/kotlin/sourceconnector/domain/batch/DefaultMessageBatchLogs.kt
package sourceconnector.domain.batch

class DefaultMessageBatchLogs(
  private val logs: MutableList<String>
) : MessageBatch<String> {

  fun add(log: String) {
    this.logs.add(log)
  }

  fun addAll(logs: MutableList<String>) {
    this.logs.addAll(logs)
  }

  override fun get(): List<String> {
    check(logs.isNotEmpty()) { "No log found" }
    return this.logs
  }
}
